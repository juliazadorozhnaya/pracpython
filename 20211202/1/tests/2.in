class C(int, metaclass=dump):
    def __init__(self, val):
        self.val = val

    def __add__(self, other):
        return C(self.val + other // 2)

    def ask(self, *, where='nowhere', what='nothing'):
        if where == 'nowhere' and what == 'nothing':
            return 'nothing to answer';
        if where == 'nowhere' and what == 'question of life':
            return 'i dont know what are you talking about'
        if where == 'here' and what == 'nothing':
            return 'nothing is nothing, even nothing is not nothing but something'
        if where == 'here' and what == 'question of life':
            return '42' if self.val == 42 else 'what?'

a = C(10)
print(a)
print(a.ask())
print(a + 10)
print(a.ask(where='here'))
print(a.ask(what='question of life'))
print(a.ask(where='here', what='question of life'))
print((a + 64).ask(where='here', what='question of life'))
