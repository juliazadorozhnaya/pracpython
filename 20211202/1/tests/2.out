__init__: (10,), {}
10
ask: (), {}
nothing to answer
__add__: (10,), {}
__init__: (15,), {}
15
ask: (), {'where': 'here'}
nothing is nothing, even nothing is not nothing but something
ask: (), {'what': 'question of life'}
i dont know what are you talking about
ask: (), {'where': 'here', 'what': 'question of life'}
what?
__add__: (64,), {}
__init__: (42,), {}
ask: (), {'where': 'here', 'what': 'question of life'}
42
